packages/kbn-router-to-openapispec/src/__snapshots__/generate_oas.test.ts.snap

```diff
@@ -632,6 +636,102 @@ Object {
 }
 `;
 
+exports[`generateOpenApiDocument Zod handles recursive schemas 1`] = `
+Object {
+  "components": Object {
+    "schemas": Object {},
```

Comment:
I would expect this field to populated with something like:

```
  "components": Object {
    "schemas": Object {
      "recursive": Object {
        "additionalProperties": false,
        "properties": Object {
          "name": Object {
            "type": "string",
          },
          "self": Object {
            "$ref": "#/components/schemas/recursive",
          },
        },
        "required": Array [
          "name",
          "self",
        ],
        "type": "object",
      },
    },    
```
I think we will need to update `$refStrategy` to something other than `none`... Probably `root`? In `packages/kbn-router-to-openapispec/src/oas_converter/zod/lib.ts`

I think we can create a follow up item to support recursive OAS schemas in Zod and exclude it from here for now. Up to you if you'd like to cover it in this PR too though.

Reply:
Covering recursive in a follow-up ticket sounds good to me.

Reply:
Ticket: https://github.com/elastic/kibana/issues/187335